tests/lit-tests/explicit-vector.ispc

```diff
@@ -0,0 +1,85 @@
+// This test checks for the simplification of induction variables for loops and efficient
+// generation of accumulator widening/arithmetic.
```

Comment:
To check that induction variable was widened no need for full original test.

Key point is that 32-bit variable is used in 64 bit context, for example as index for load operation on 64 bit target. I.e. something like
```c
uniform float result = 0;
for (uniform int i = 0; i < Num; i += N) {
    result += A[i];
}
```
is sufficient and checks exactly the key point. Minimizing the test makes it more robust.

Also, I'm not sure that checking after specific phase is needed - checking on the last phase makes more sense as it checks that this transformation happened / not happened at all.

Reply:
See new test for indvars: `tests/lit-tests/ind-var-simplifiy.ispc`.

Reply:
I kept this lit test with the portion which explicitly checks the x86 assembly requested in Issue #2460.

Reply:
If you keep it for the sake of testing original issue, then
- rename to `2460_indvar_promotion.ispc`
- I understand checking for `cdqe` and `movsxd`.
- what is `vmovmask`, I think there's no such instruction. There are `vmovmsk*` and `vmaskmov*`. But still I don't understand what is the problem with them in context of induction variable promotion.
- `vpinsrd` - same question - what is the problem with it in the context of induction variable promotion?

Reply:
The request from Jeff explicitly mentions these instructions in the issue https://github.com/ispc/ispc/issues/2460 . They (e.g. vpinsrd) are generated as a consequence of the failed induction variable promotion, and should also not be present if the optimization is done correctly. Good catch on `vmaskmov` typo - will fix.

Reply:
I think `vmovmsk` and `vpinsrd` are not directly related to sign extension issue. Am I missing something?